lib/controllers/messages.ts

```diff
@@ -276,21 +257,24 @@ async function getMessageNotificationStatuses(
     // collect the statuses of all channels
     const channelStatusesPromises = Object.keys(NotificationChannelEnum)
       .map(k => NotificationChannelEnum[k as NotificationChannelEnum])
-      .map(channel => ({
+      .map(async channel => ({
         channel,
-        status: getChannelStatus(channel)
+        status: await getChannelStatus(channel)
```

Comment:
I believe you don't want an await here, the map is synchronous and creates an array of promises, the await should happen just on Promise.all